core/includes/config.inc

```diff
@@ -288,28 +288,60 @@ function config_install_default_config($project, $config_name = NULL) {
     }
   }
   $project_config_dir = $project_path . '/config';
+  $storage = new ConfigFileStorage($project_config_dir);
+
+  $default_config_names = config_get_default_config_names($project);
+  foreach ($default_config_names as $default_config_name) {
+    // Load config data into the active store and write it out to the
+    // file system in the Backdrop config directory. Note the config name
+    // needs to be the same as the file name WITHOUT the extension.
+    if (is_null($config_name) || $default_config_name === $config_name) {
+      $data = $storage->read($default_config_name);
+      $config = config($default_config_name);
+      // We only create new configs, and do not overwrite existing ones.
+      if ($config->isNew()) {
+        $config->setData($data);
+        module_invoke_all('config_create', $config);
+        $config->save();
+      }
+    }
+  }
+}
+
+/**
+ * Gets the name of config files provided by a module or theme.
+ *
+ * @param $project
+ *   The name of the project for which config names should be returned.
+ *
+ * @return string[]
+ *   An unindexed array of config names, without the .json extension.
+ *
+ * @since 1.31.0 Function added.
+ */
+function config_get_default_config_names($project) {
```

Comment:
I separated this out from `config_install_default_config()` since I needed this same functionality in the test coverage. See `ModuleTestCase::assertModuleConfigExists()`.